### ISSUE TITLE
It is great, but not support Chinese and Japanese

### ISSUE THREAD

**Issue by gliese667c** [kyle-n/HighlightedTextView#8]
**Describe the bug**
It is great, but not support Chinese and Japanese.

**To Reproduce**
**Just add a Chinese or Japanese keyboard**, type characters always are English.

**Comment by kyle-n** [comment]
Thank you for reporting this. I haven't done much with localization. I will investigate this and see how to address this.

**Comment by kyle-n** [comment]
I am not sure how to reproduce this, I will need more information. 

I used this test project to try to reproduce the bug:

```swift
import SwiftUI
import HighlightedTextEditor
import UIKit

@main
struct TestingiOSApp: App {
    var body: some Scene {
        WindowGroup {
            ContentView()
        }
    }
}

let x = try! NSRegularExpression(pattern: "但", options: [])

struct ContentView: View {
    @State private var text: String = "abc\n//comment\n//comment 2\n\nnot comment"
    @State private var toggler: Bool = false
    
    var body: some View {
        VStack {
            Button("Toggle") { toggler.toggle() }
            HighlightedTextEditor(text: $text, highlightRules: [
                HighlightRule(pattern: x, formattingRules: [
                    TextFormattingRule(key: .foregroundColor, value: UIColor.blue),
                    TextFormattingRule(key: .font, value: UIFont.boldSystemFont(ofSize: 99))
                ])
            ])
        }
    }
}
```

I used the Chinese Traditional Pinyin keyboard to type the 但 character in HighlightedTextEditor. It was highlighted with the blue color and large font size, as expected. I was also able to highlight Japanese characters using the Japanese Kana keyboard. These results are using the latest version of HLTE on iPadOS 14.2.

When I typed Chinese or Japanese characters, those characters were rendered in HLTE, not English characters. 

To investigate this bug, I would like more information. Could you provide:

- The exact keyboard, OS, and OS version where you saw the bug
- UIKit or AppKit? Sounds like UIKit
- Steps to reproduce this?
- Expected result
- actual result

**Comment by gliese667c** [comment]
The problem is HLTE behaves differently from the native Text Editor when using Chinese and some other languages. In the test case, I use the same keyboard (you can use any kind of Chinese keyboard) for typing, they were totally different. The expected result is the native TextEditor does.
![ezgif com-gif-maker](https://user-images.githubusercontent.com/11246312/99011179-76912080-2586-11eb-9532-0145c0551bf6.gif)
The code I tested.
```swift
import SwiftUI
import HighlightedTextEditor
import UIKit

@main
struct TestingiOSApp: App {
    var body: some Scene {
        WindowGroup {
            ContentView()
        }
    }
}

let x = try! NSRegularExpression(pattern: "但", options: [])

struct ContentView: View {
    @State private var text: String = "This is a HLTE TextEditor"
    @State private var toggler: Bool = false
    @State private var nativeTextEditorText:String = "This is a native TextEditor."
    var body: some View {
        VStack {
            Button("Toggle") { toggler.toggle() }
            HighlightedTextEditor(text: $text, highlightRules: [
                HighlightRule(pattern: x, formattingRules: [
                    TextFormattingRule(key: .foregroundColor, value: UIColor.blue),
                    TextFormattingRule(key: .font, value: UIFont.boldSystemFont(ofSize: 99))
                ])
            ])
            TextEditor(text: $nativeTextEditorText)
        }
    }
}
```

**Comment by kyle-n** [comment]
Thank you for the GIF! That was perfect for displaying the problem. 

I was able to reproduce the problem. Certain keyboards (such as Chinese, Traditional - Pinyin and Japanese - Kana, the two I tested) use "marked" text to indicate a word still in the process of being typed in a UITextView. When the English characters are replaced by a Chinese character, it is no longer "marked," but "committed." [TIL](https://stackoverflow.com/questions/12674949/how-to-detect-user-is-mid-way-inputting-with-an-input-method/).

I have pushed a commit to the `develop` branch with code that should fix this problem. It appears to work for me, but as someone who does not regularly use these keyboards, I would like to make sure. 

Could you pull down the latest version of HLTE's `develop` branch and confirm that it works with Chinese or Japanese keyboards?

**Comment by gliese667c** [comment]
I actually tried to find out what the problem was, but I had no idea, so I can not fix it.

**Comment by kyle-n** [comment]
I have merged a fix for the issue into `main` It has been released as version 1.4.1.
